SwiftyStoreKit/InAppReceipt.swift

```diff
@@ -235,5 +253,73 @@ internal class InAppReceipt {
             }
             task.resume()
     }
+
+    /**
+     *  Verify the purchase of a product in a receipt
+     *  - Parameter productId: the product id of the purchase to verify
+     *  - Parameter inReceipt: the receipt to test in
+     *  - Parameter validUntil: the expires date of the subscription must be valid until this date. If nil, no verification
+     */
+    class func verifyPurchase(
+      productId productId: String,
+      inReceipt receipt: ReceiptInfo,
+      purchaseType: SwiftyStoreKit.PurchaseType
+    ) -> SwiftyStoreKit.VerifyPurchaseResult {
+      
+        // Get all receipts
+        guard let allReceipts = receipt["receipt"]?["in_app"] as? [ReceiptInfo] else {
+            return .NotPurchased
+        }
+      
+        // Filter receipts with matching product id
+        let receiptsMatchingProductId = allReceipts
+            .filter { (receipt) -> Bool in
+                let product_id = receipt["product_id"] as? String
+                return product_id == productId
+        }
+      
+        // Verify that at least one receipt has the right product id
+        guard receiptsMatchingProductId.count >= 1 else {
+            return .NotPurchased
+        }
+      
+        switch purchaseType {
+            // Do not need to verify the expiration date
+            case .Consumable, .NonConsumable, .FreeSubscription:
+                return .Purchased(expiresDate: nil)
+            case .AutomaticallyRenewableSubscription(let validUntilDate):
+                return verifyPurchase(receiptsMatchingProductId, validUntilDate: validUntilDate)
+            case .NonRenewingSubscription(let validUntilDate):
+                return verifyPurchase(receiptsMatchingProductId, validUntilDate: validUntilDate)
+        }
+    }
     
+    private class func verifyPurchase(receiptsMatchingProductId: [ReceiptInfo], validUntilDate date: NSDate) -> SwiftyStoreKit.VerifyPurchaseResult {
+        // Return the expires dates sorted desc
+        let expiresDates = receiptsMatchingProductId
+            .map { (receipt) -> NSDate in
+                let expires_date = receipt["expires_date_ms"] as? NSString
+                let expires_date_double = (expires_date?.doubleValue ?? 0.0) / 1000
+                return NSDate(timeIntervalSince1970: expires_date_double)
```

Comment:
Is there a way that we can more gracefully handle the case where the `expires_date` value is missing and not return "1 January 1970"?